src/publicTypes.ts

```diff
@@ -25,6 +25,15 @@ export enum NamingServiceName {
 }
 export type ResolutionMethod = NamingServiceName | 'UDAPI';
 
+export function hasProvider(obj: any): obj is { provider: Provider } {
+  return obj && !!obj.provider;
+}
+
+export type AutoNetworkConfigs = {
+  ens?: { url: string } | { provider: Provider },
```

Comment:
@JohnnyJumper Do we need to mark properties as optional? In this case calling `Resolution.autoNetwork` is equal to `new Resolution()`

Reply:
I marked them as optional to not force a user to configure both CNS and ENS. They should be able to choose which one they want. if no configurations were provided I think it is normal to use the default settings instead or maybe it is better to just throw an error?